src/lib/eina/eina_thread_win32.c

```diff
@@ -0,0 +1,213 @@
+{
+   Eina_Bool ret;
+   Eina_win32_thread_attr thread_attr;
+
+   SECURITY_ATTRIBUTES sec_attributes;
+
+   sec_attributes.nLength = sizeof(SECURITY_ATTRIBUTES);
+   sec_attributes.lpSecurityDescriptor = NULL;
+   sec_attributes.bInheritHandle = EINA_TRUE;
+
+   thread_attr.lpThreadAttributes = &sec_attributes;
+
+   thread_attr.dwStackSize = 4000;
+   thread_attr.dwCreationFlags = 0;
+
+   LPDWORD threadID;
+
+   Eina_win32_thread_func thread_func;
+   Eina_Thread_Call *c = (Eina_Thread_Call *)(data);
+
+   thread_func.func = func;
+   thread_func.data = data;
+
+   *t = (HANDLE)CreateThread(thread_attr.lpThreadAttributes, thread_attr.dwStackSize, &_eina_thread_func, &thread_func, thread_attr.dwCreationFlags, threadID);
+
+   _eina_thread_set_priority(c->prio, t);
+
+   ret = (*t != NULL) ? EINA_TRUE : EINA_FALSE;
+
+   if (affinity >= 0 && ret)
+    {
+#ifdef EINA_HAVE_PTHREAD_AFFINITY
+      cpu_set_t cpu;
+      CPU_ZERO(&cpu);
+      CPU_SET(affinity, &cpu);
+      SetThreadAffinityMask(*t, (DWORD_PTR *)&cpu);
+#else
+      SetThreadAffinityMask(*t, (DWORD_PTR *)&affinity);
+#endif
+    }
+
+   return ret;
+}
+
+inline Eina_Bool
+_eina_thread_equal(Eina_Thread t1, Eina_Thread t2)
+{
+   DWORD t1_thread_id = GetThreadId((HANDLE)t1);
+   DWORD t2_thread_id = GetThreadId((HANDLE)t2);
+
+   return (t1_thread_id == t2_thread_id) ? EINA_TRUE : EINA_FALSE;
+}
+
+inline Eina_Thread
+_eina_thread_self(void)
+{
+   return (Eina_Thread)GetCurrentThread();
+}
+
+HRESULT _eina_thread_set_name_win32(Eina_Thread thread, char *buf)
+{
+   return SetThreadDescription((HANDLE)thread, (PCWSTR)buf);
+}
+
+Eina_Bool _eina_thread_cancel(Eina_Thread thread)
+{
+   LPDWORD lpExitCode;
+   Eina_Bool success = GetExitCodeThread((HANDLE)thread, lpExitCode);
+
+   ExitThread(*lpExitCode);
+   return success;
+}
+
+inline void _eina_sched_prio_drop(void)
+{
+   Eina_Thread pthread_id;
+   int sched_priority;
+
+   pthread_id = eina_thread_self();
+
+   sched_priority = GetThreadPriority((HANDLE)pthread_id);
+
+   if (EINA_UNLIKELY(sched_priority == THREAD_PRIORITY_TIME_CRITICAL))
+    {
+      sched_priority -= RTNICENESS;
+
+      /* We don't change the policy */
+      if (sched_priority < 1)
+       {
+         EINA_LOG_INFO("RT prio < 1, setting to 1 instead");
+         sched_priority = 1;
+       }
+      if (!SetThreadPriority((HANDLE)pthread_id, sched_priority))
+       {
+         EINA_LOG_ERR("Unable to query sched parameters");
+       }
+   }
+   else
+   {
+      sched_priority += NICENESS;
+
+      /* We don't change the policy */
+      if (sched_priority > THREAD_PRIORITY_TIME_CRITICAL)
+       {
+         EINA_LOG_INFO("Max niceness reached; keeping max (THREAD_PRIORITY_TIME_CRITICAL)");
+         sched_priority = THREAD_PRIORITY_TIME_CRITICAL;
+       }
+      if (!SetThreadPriority((HANDLE)pthread_id, sched_priority))
+       {
+         EINA_LOG_ERR("Unable to query sched parameters");
+       }
+   }
+}
```

Comment:
Why is a sched function being defined here and not in an eina_sched_win32.c file?

Reply:
now it is